### PR TITLE
feat(authentication): add sign-in and sign-up forms

### PR DESCRIPTION
**Pull request by avramenkop**

### CODE REVIEW

File: swapi/.vscode/extensions.json
Comment (otecMuxah) [inline swapi/.vscode/extensions.json:1]: .vscode folder should be ignored by gitignore

File: swapi/src/app/app.component.ts
Comment (otecMuxah) [inline swapi/src/app/app.component.ts:1]: i guess it is unused component, and everything that is not being used , should be removed.

File: swapi/src/app/app.module.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import { MatFormFieldModule } from "@angular/material/form-field";
+import { MatIconModule } from "@angular/material/icon";
+import { MatDialogModule } from "@angular/material/dialog";
+import { MatButtonModule } from "@angular/material/button";
+import { MatInputModule } from "@angular/material/input";
+
+const materialModules = [
+  MatFormFieldModule,
+  MatIconModule,
+  MatDialogModule,
```
Comment (otecMuxah) [inline swapi/src/app/app.module.ts]: looks unused

File: swapi/src/app/app.module.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  MatButtonModule,
+  MatInputModule
+];
+
+@NgModule({
+  declarations: [
+    AppComponent,
```
Comment (otecMuxah) [inline swapi/src/app/app.module.ts:24]: unused

File: swapi/src/app/app.module.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  ],
+  imports: [
+    BrowserModule,
+    AppRoutingModule,
+    BrowserAnimationsModule,
+    ReactiveFormsModule,
+    ...materialModules,
+    FormsModule
+  ],
+  providers: [],
```
Comment (otecMuxah) [inline swapi/src/app/app.module.ts]: this is meaningless, please remove

File: swapi/src/app/sign-in/sign-in.component.html [hunk 4]
```diff
@@ -0,0 +1,69 @@
+<div class="container">
+  <form #loginForm="ngForm" (ngSubmit)="saveDetails()">
+    <mat-form-field>
+      <mat-label>Email:</mat-label>
+      <input
+        #email="ngModel"
+        matInput
+        type="text"
+        inputmode="email"
```
Comment (otecMuxah) [inline swapi/src/app/sign-in/sign-in.component.html:9]: not supported in safari

File: swapi/src/app/sign-up/sign-up.component.html [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+        matInput
+        [type]="hideConfirmEmail ? 'password' : 'text'"
+        formControlName="confirmPassword">
+      <button
+        type="button"
+        mat-icon-button
+        matSuffix
+        (click)="hideConfirmEmail = !hideConfirmEmail"
+        [attr.aria-label]="'Hide password'"
+        [attr.aria-pressed]="hideConfirmEmail">
```
Comment (otecMuxah) [inline swapi/src/app/sign-up/sign-up.component.html:59]: cool stuff

File: swapi/src/app/sign-up/utils/sign-up.utils.ts [hunk 6]
```diff
@@ -0,0 +1,8 @@
+import { AbstractControl, ValidationErrors, ValidatorFn } from "@angular/forms";
+
+export const passwordMatchValidator: ValidatorFn = (control: AbstractControl): ValidationErrors | null => {
+  const password = control.get('password');
+  const confirmPassword = control.get('confirmPassword');
+
+  return password && confirmPassword && password.value === confirmPassword.value ? { identityRevealed: true } : null;
```
Comment (otecMuxah) [inline swapi/src/app/sign-up/utils/sign-up.utils.ts]: not sure i saw where do you use identityRevealed

Comment (otecMuxah) [reply swapi/src/app/sign-up/utils/sign-up.utils.ts]: also not sure i saw, you display that passwords are not matched error to user

**Review by otecMuxah — CHANGES_REQUESTED** [review]

File: swapi/src/app/sign-up/sign-up.component.ts [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+  templateUrl: './sign-up.component.html',
+  styleUrls: ['./sign-up.component.scss']
+})
+export class SignUpComponent {
+  private fb = inject(FormBuilder);
+
+  signUpForm = this.fb.group({
+    email: ['', {validators: [Validators.required, Validators.email]}],
+    password: ['', [Validators.required, Validators.minLength(6)]],
+    confirmPassword: ['', [Validators.required]],
```
Comment (otecMuxah) [inline swapi/src/app/sign-up/sign-up.component.ts]: why no minLength here ?

File: swapi/src/app/app.component.ts
Comment (avramenkop) [reply swapi/src/app/app.component.ts:1]: What did you mean exactly? 'title' prop is really unused. but in the template there is a router-outlet.
should I move router-outlet to the controller file and delete the template?

File: swapi/src/app/app.module.ts (hunk 1 above)
Comment (avramenkop) [reply swapi/src/app/app.module.ts]: these modules are used in the both forms. material directives won't work if delete

File: swapi/src/app/app.module.ts (hunk 3 above)
Comment (avramenkop) [reply swapi/src/app/app.module.ts]: is the dedicated materialModules array meaningless?

File: swapi/src/app/sign-up/sign-up.component.html (hunk 5 above)
Comment (avramenkop) [reply swapi/src/app/sign-up/sign-up.component.html:59]: is it really good approach? please resolve in this case

File: swapi/src/app/sign-up/sign-up.component.ts (hunk 7 above)
Comment (avramenkop) [reply swapi/src/app/sign-up/sign-up.component.ts]: i guess it's not crucial to check `confirmPassword` minLength. However `password` and `confirmPassword` match is required

File: swapi/src/app/app.module.ts (hunk 2 above)
Comment (avramenkop) [reply swapi/src/app/app.module.ts:24]: I have no idea what does it mean

File: swapi/src/app/sign-up/utils/sign-up.utils.ts (hunk 6 above)
Comment (avramenkop) [reply swapi/src/app/sign-up/utils/sign-up.utils.ts]: I'm going to use the code in `password` and `confirmPassword` fields cross validation. just haven't added the error message

File: swapi/src/app/app.component.ts
Comment (otecMuxah) [reply swapi/src/app/app.component.ts:1]: Try to delete this file and see what happens, you do not use this file as well as router outlet

File: swapi/src/app/app.module.ts (hunk 1 above)
Comment (otecMuxah) [reply swapi/src/app/app.module.ts]: I left comment under matDialog module, are you using matDialog somewhere ?

File: swapi/src/app/app.module.ts (hunk 2 above)
Comment (otecMuxah) [reply swapi/src/app/app.module.ts:24]: It says AppComponent is unused

File: swapi/src/app/app.module.ts (hunk 3 above)
Comment (otecMuxah) [reply swapi/src/app/app.module.ts]: Wtf are you talking about ??? You left empty providers field in code , which makes no sense at all

File: swapi/src/app/sign-up/utils/sign-up.utils.ts (hunk 6 above)
Comment (otecMuxah) [reply swapi/src/app/sign-up/utils/sign-up.utils.ts]: This validator has reverse logic it works exactly opposite way compared to what we want here

File: swapi/src/app/sign-up/sign-up.component.ts (hunk 7 above)
Comment (otecMuxah) [reply swapi/src/app/sign-up/sign-up.component.ts]: it is wrong to have the same fields with a different set of validators

File: swapi/.vscode/extensions.json
Comment (avramenkop) [reply swapi/.vscode/extensions.json:1]: done

File: swapi/src/app/app.module.ts (hunk 1 above)
Comment (avramenkop) [reply swapi/src/app/app.module.ts]: deleted

File: swapi/src/app/sign-in/sign-in.component.html (hunk 4 above)
Comment (avramenkop) [reply swapi/src/app/sign-in/sign-in.component.html:9]: works on iOS 16.1.1 
IMG_0923

File: swapi/src/app/sign-up/utils/sign-up.utils.ts (hunk 6 above)
Comment (avramenkop) [reply swapi/src/app/sign-up/utils/sign-up.utils.ts]: updated

File: swapi/src/app/sign-up/sign-up.component.ts (hunk 7 above)
Comment (avramenkop) [reply swapi/src/app/sign-up/sign-up.component.ts]: done

File: swapi/src/app/sign-in/sign-in.component.html (hunk 4 above)
Comment (otecMuxah) [reply swapi/src/app/sign-in/sign-in.component.html:9]: not ios , but macos , check mdn ar caniuse

**Review by otecMuxah — APPROVED** [review]

**Pull request merged**